server/src/main/java/org/elasticsearch/search/retriever/rankdoc/RankDocsQuery.java

```diff
@@ -242,14 +261,26 @@ public int hashCode() {
      * @param sources      The original queries that were used to compute the top documents
      * @param queryNames   The names (if present) of the original retrievers
      * @param onlyRankDocs Whether the query should only match the provided rank docs
+     * @param minScore     The minimum score threshold for documents to be included in total hits.
+     *                     This can be set to any value including 0.0f to filter out documents with scores below the threshold.
+     *                     Note: This is separate from the special handling of 0 scores. Documents with a score of exactly 0
+     *                     will always be assigned DEFAULT_MIN_SCORE internally to differentiate them from filtered matches,
+     *                     regardless of the minScore value.
      */
-    public RankDocsQuery(IndexReader reader, RankDoc[] rankDocs, Query[] sources, String[] queryNames, boolean onlyRankDocs) {
+    public RankDocsQuery(
+        IndexReader reader,
+        RankDoc[] rankDocs,
+        Query[] sources,
+        String[] queryNames,
+        boolean onlyRankDocs,
+        float minScore
+    ) {
         assert sources.length == queryNames.length;
         // clone to avoid side-effect after sorting
         this.docs = rankDocs.clone();
         // sort rank docs by doc id
         Arrays.sort(docs, Comparator.comparingInt(a -> a.doc));
-        this.topQuery = new TopQuery(docs, sources, queryNames, findSegmentStarts(reader, docs), reader.getContext().id());
+        this.topQuery = new TopQuery(docs, sources, queryNames, findSegmentStarts(reader, docs), reader.getContext().id(), minScore);
```

Comment:
Flagging this to verify - we pass in  min score to top docs here, but we don't do this in the new PR. Was this testing, or is this logic that's missing?